flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/workflow/scheduler/EmbeddedQuartzScheduler.java

```diff
@@ -214,16 +248,230 @@ private void checkJobExists(JobKey jobKey, String errorMsg)
+                                EmptyRequestBody.getInstance())
+                        .get();
+            }
+
+            public OperationHandle refreshMaterializedTable(
+                    SessionHandle sessionHandle,
+                    String materializedTableIdentifier,
+                    boolean ignoreFailures,
+                    String schedulerTime,
+                    Map<String, String> dynamicOptions,
+                    Map<String, String> staticPartitions,
+                    Map<String, String> executionConfig)
+                    throws Exception {
+
+                RefreshMaterializedTableRequestBody requestBody =
+                        new RefreshMaterializedTableRequestBody(
+                                true,
+                                schedulerTime,
+                                dynamicOptions,
+                                staticPartitions,
+                                executionConfig);
+                RefreshMaterializedTableHeaders headers =
+                        RefreshMaterializedTableHeaders.getInstance();
+                RefreshMaterializedTableParameters parameters =
+                        new RefreshMaterializedTableParameters(
+                                sessionHandle, materializedTableIdentifier);
+
+                RefreshMaterializedTableResponseBody responseBody =
+                        restClient
+                                .sendRequest(address, port, headers, parameters, requestBody)
+                                .get();
+
+                return new OperationHandle(UUID.fromString(responseBody.getOperationHandle()));
+            }
+
+            public List<RowData> fetchOperationAllResults(
+                    SessionHandle sessionHandle, OperationHandle operationHandle) throws Exception {
+                Long token = 0L;
+                List<RowData> results = new ArrayList<>();
+                while (token != null) {
+                    FetchResultsResponseBody responseBody =
+                            fetchOperationResults(sessionHandle, operationHandle, token);
+                    if (responseBody instanceof NotReadyFetchResultResponse) {
+                        continue;
+                    }
+                    responseBody.getNextResultUri();
+                    results.addAll(responseBody.getResults().getData());
+                    token = SqlGatewayRestEndpointUtils.parseToken(responseBody.getNextResultUri());
+                }
+                return results;
+            }
+
+            public FetchResultsResponseBody fetchOperationResults(
+                    SessionHandle sessionHandle, OperationHandle operationHandle, Long token)
+                    throws Exception {
+                FetchResultsMessageParameters fetchResultsMessageParameters =
+                        new FetchResultsMessageParameters(
+                                sessionHandle, operationHandle, token, RowFormat.PLAIN_TEXT);
```

Comment:
Why it is not `RowFormat.JSON`?

Reply:
changed to RowFormat.JSON.  JsonRowFormat will retain the original field and its type.